In [1]:
import pandas as pd

In [4]:
# Read data in
df = pd.read_csv('input.csv')
df.head()

Set #         Name Rarity  Normal/Holo  Rev  Rev - M Price - M
0    #1    Bulbasaur      C            7  2.0      1.0    $22.64
1    #2      Ivysaur      U            5  1.0      NaN    $15.00
2    #3  Venusaur ex     RR            2  NaN      NaN       NaN
3    #4   Charmander      C            8  1.0      NaN    $33.00
4    #5   Charmeleon      U            4  NaN      1.0    $14.25

In [5]:
# fill nulls with 0
df.fillna(value=0, inplace=True)

In [6]:
# change column type to ints
df['Rev'] = df['Rev'].astype('int64')
df['Rev - M'] = df['Rev - M'].astype('int64')

In [7]:
# 1st condition, remove ex
# 2nd condition, 
filtered_df = df[~df['Name'].str.contains('ex')].loc[0:164]
filtered_df.head()

Set #        Name Rarity  Normal/Holo  Rev  Rev - M Price - M
0    #1   Bulbasaur      C            7    2        1    $22.64
1    #2     Ivysaur      U            5    1        0    $15.00
3    #4  Charmander      C            8    1        0    $33.00
4    #5  Charmeleon      U            4    0        1    $14.25
6    #7    Squirtle      C            7    1        0    $21.86

In [8]:
# As of 7/30/2024, these are the cards I still need to complete the 151 sv2a set
missing_rev_df = filtered_df[filtered_df['Rev'] == 0]
missing_rev_df

Set #        Name Rarity  Normal/Holo  Rev  Rev - M Price - M
4      #5  Charmeleon      U            4    0        1    $14.25
18    #19     Rattata      C            6    0        1    $10.75
46    #47    Parasect      U            2    0        1    $10.50
117  #118     Goldeen      C           10    0        1    $10.14
148  #149   Dragonite      R            5    0        1    $41.02

In [11]:
mast_df = filtered_df[filtered_df['Rev - M'] == 0]
mast_df.head()

Set #        Name Rarity  Normal/Holo  Rev  Rev - M Price - M
1     #2     Ivysaur      U            5    1        0    $15.00
3     #4  Charmander      C            8    1        0    $33.00
6     #7    Squirtle      C            7    1        0    $21.86
9    #10    Caterpie      C            8    1        0    $10.70
10   #11     Metapod      C           10    2        0    $11.27

# Cards Needed:

### Reverse (Pokeball) Holos:
* \#1 Bulbasaur
* \#5 Charmeleon
* \#19 Rattata
* \#47 Parasect
* \#118 Goldeen
* \#149 Dragonite

### Masterball Holos:
66 left